modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache

```diff
@@ -0,0 +1,87 @@
+import {Http, Headers, RequestOptionsArgs, Response, URLSearchParams} from 'angular2/http';
+import {Injectable} from 'angular2/core';
+import {Observable} from 'rxjs/Observable';
+import * as model from "../model/model.d.ts"
+
+/* tslint:disable:no-unused-variable member-ordering */
+
+{{#operations}}
+'use strict';
+
+{{#description}}
+/**
+ * {{&description}}
+ */
+{{/description}}
+@Injectable()
+export class {{classname}} {
+    protected basePath = '{{basePath}}';
+    public defaultHeaders : Headers = new Headers();
+
+    constructor(protected http: Http, basePath: string) {
+        if (basePath) {
+            this.basePath = basePath;
+        }
+    }
+
+{{#operation}}
+    /**
+     * {{summary}}
+     * {{notes}}
+     {{#allParams}}* @param {{paramName}} {{description}}
+     {{/allParams}}*/
+    public {{nickname}} ({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}extraHttpRequestParams?: any ) : Observable<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}{}{{/returnType}}> {
+        const path = this.basePath + '{{path}}'{{#pathParams}}
+            .replace('{' + '{{baseName}}' + '}', String({{paramName}})){{/pathParams}};
+
+        let queryParameters: any = ""; // This should probably be an object in the future
+        let headerParams = this.defaultHeaders;
+{{#hasFormParams}}
+        let formParams = new URLSearchParams();
+
+{{/hasFormParams}}
+{{#allParams}}
+{{#required}}
+        // verify required parameter '{{paramName}}' is set
+        if (!{{paramName}}) {
+            throw new Error('Missing required parameter {{paramName}} when calling {{nickname}}');
+        }
+{{/required}}
+{{/allParams}}
+{{#queryParams}}
+        if ({{paramName}} !== undefined) {
+            queryParameters['{{baseName}}'] = {{paramName}};
+        }
+
+{{/queryParams}}
+{{#headerParams}}
+            headerParams.set('{{baseName}}', {{paramName }});
+
+{{/headerParams}}
+{{#hasFormParams}}
+        headerParams.set('Content-Type', 'application/x-www-form-urlencoded');
+
+{{/hasFormParams}}
+{{#formParams}}
+        formParams['{{baseName}}'] = {{paramName}};
+
+{{/formParams}}
+        let requestOptions: RequestOptionsArgs = {
+            method: '{{httpMethod}}',
+            headers: headerParams,
+            search: queryParameters
+        };
+        {{#bodyParam}}
+        requestOptions.body = JSOG.stringify({{paramName}});
```

Comment:
@Vrolijkx seems like you have dropped the `import JSOG` statement, did you forget to swap it here?